goalgorilla/open_social
# Issue #3208582 by ronaldtebrake: As a LU when editing a flexible group I want my chosen join method settings to be respected

## Problem
When I create a flexible group and choose group visibility, content visibility and join methods all works well.
Whenever I edit that group, the join method automatically jumps to the "request to enroll" without the user selecting it, it seems the Drupal states are all wrong.

## Solution
Remove the states logic and redo everything with javascript.

## Issue tracker
https://www.drupal.org/project/social/issues/3208582
https://www.drupal.org/project/social/issues/3204470

## How to test
1. Create a flexible group with everything set to Public
2. Edit that group and see it switches the join method to "request to enroll"
